ciao-controller/instance.go

```diff
@@ -107,14 +108,20 @@ func newInstance(ctl *controller, tenantID string, workload *types.Workload,
 func (i *instance) Add() error {
 	ds := i.ctl.ds
 	var err error
-	if i.CNCI == false {
-		err = ds.AddInstance(&i.Instance)
-	} else {
-		err = ds.AddTenantCNCI(i.TenantID, i.ID, i.MACAddress)
-	}
+	err = ds.AddInstance(&i.Instance)
```

Comment:
Nothing do with the code you've modified but if something later fails we've got an inconsistent state with an instance in the datastore but returned an error.